### PR TITLE
Fix Table text wrapping

### PR DESCRIPTION
**Pull request by guillim**
As discovered during the last release, text fields in the table were wrapped. This PR fixes this unwanted behaviour

Current :
Screenshot 2024-11-22 at 14 17 42

Wanted : 
One line ellipsed
Screenshot 2024-11-22 at 14 20 46

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/data-model/fields/preview/components/SettingsDataModelFieldPreview.tsx [hunk 1]
```diff
@@ -140,6 +140,7 @@ export const SettingsDataModelFieldPreview = ({
               defaultValue: fieldMetadataItem.defaultValue,
             },
             hotkeyScope: 'field-preview',
+            wrap: true,
```
[the remark this answers is not in the document]
Comment (guillim) [reply packages/twenty-front/src/modules/settings/data-model/fields/preview/components/SettingsDataModelFieldPreview.tsx:143]: should be true for Preview

File: packages/twenty-front/src/modules/object-record/record-field/contexts/FieldContext.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ export type GenericFieldContextType = {
   maxWidth?: number;
   isCentered?: boolean;
   overridenIsFieldEmpty?: boolean;
+  wrap?: boolean;
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/contexts/FieldContext.ts:34]: allowDisplayWrap

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/TextFieldDisplay.tsx [hunk 3]
```diff
@@ -3,11 +3,17 @@ import { isFieldText } from '@/object-record/record-field/types/guards/isFieldTe
...
   const displayedMaxRows = isFieldText(fieldDefinition)
     ? fieldDefinition.metadata?.settings?.displayedMaxRows
     : 1;
 
-  return <TextDisplay text={fieldValue} displayedMaxRows={displayedMaxRows} />;
+  return (
+    <TextDisplay
+      text={fieldValue}
+      displayedMaxRows={displayedMaxRows}
+      wrap={wrap}
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/TextFieldDisplay.tsx:16]: I would use allowDisplayWrap (or wrap) here to compute displayedMaxRows directly, do we need two props?

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/hooks/useTextFieldDisplay.ts [hunk 4]
```diff
@@ -2,10 +2,13 @@ import { useContext } from 'react';
...
 import { useRecordFieldValue } from '@/object-record/record-store/contexts/RecordFieldValueSelectorContext';
 
+import { useRecordInlineCellContext } from '@/object-record/record-inline-cell/components/RecordInlineCellContext';
 import { FieldContext } from '../../contexts/FieldContext';
 
 export const useTextFieldDisplay = () => {
-  const { recordId, fieldDefinition, hotkeyScope } = useContext(FieldContext);
+  const { recordId, fieldDefinition, hotkeyScope, wrap } =
+    useContext(FieldContext);
+  const inlineCellContext = useRecordInlineCellContext();
```
[the remark this answers is not in the document]
Comment (guillim) [reply packages/twenty-front/src/modules/object-record/record-field/meta-types/hooks/useTextFieldDisplay.ts:11]: Agree. Did not manage to get something else working, but happy to find out something better with you

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/TextFieldDisplay.tsx (hunk 3 above)
Comment (guillim) [reply packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/TextFieldDisplay.tsx]: In a Table mode, if we have displayMaxRows = 5 then, unless wrap = false the table will still be mulitline. You see some other way to calculate that ?

**Comment by guillim** [comment]
should be good now. Much more concise and clearer
Screenshot 2024-11-22 at 16 42 48 Screenshot 2024-11-22 at 16 43 05

Screenshot 2024-11-22 at 16 42 15

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**